### PR TITLE
Remove all unnecessary artefacts and adapt common conventions for Pimcore bundles

### PR DESCRIPTION
**Pull request by mike4git**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,8 +25,8 @@ the following:
 ```php
 <?php
 
-use Neusta\Pimcore\EditorConfigBundle\Document\Base\HasDialogBox;
-use Neusta\Pimcore\EditorConfigBundle\Document\EditableDialogBox\DialogBoxBuilder;
+use Neusta\Pimcore\AreabrickConfigBundle\src\Document\Base\HasDialogBox;
+use Neusta\Pimcore\AreabrickConfigBundle\src\Document\EditableDialogBox\DialogBoxBuilder;
```
Comment (jdreesen) [inline README.md:29]: This is not right

```suggestion
use Neusta\Pimcore\AreabrickConfigBundle\Document\Base\HasDialogBox;
use Neusta\Pimcore\AreabrickConfigBundle\Document\EditableDialogBox\DialogBoxBuilder;
```

File: phpunit.xml.dist [hunk 2]
```diff
@@ -1,44 +1,26 @@
...
-    bootstrap="vendor/autoload.php"
-    failOnRisky="true"
-    failOnWarning="true"
->
-<!--    xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"-->
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"
+        backupGlobals="false"
+        colors="true"
+        bootstrap="tests/bootstrap.php"
```
Comment (jdreesen) [inline phpunit.xml.dist:7]: There is no `tests/bootstrap.php`

```suggestion
        bootstrap="vendor/autoload.php"
```

File: phpunit.xml.dist [hunk 3]
```diff
@@ -1,44 +1,26 @@
...
-    xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/8.5/phpunit.xsd"
-    backupGlobals="false"
-    colors="true"
-    bootstrap="vendor/autoload.php"
-    failOnRisky="true"
-    failOnWarning="true"
->
-<!--    xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"-->
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"
```
Comment (jdreesen) [inline phpunit.xml.dist:4]:
```suggestion
        xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
```

File: composer.json [hunk 4]
```diff
@@ -33,29 +33,34 @@
...
       "bundles": [
-        "Neusta\\Pimcore\\EditorConfigBundle\\NeustaPimcoreEditorConfigBundle"
+        "Neusta\\Pimcore\\AreabrickConfigBundle\\NeustaPimcoreAreabrickConfigBundle"
       ]
     }
   },
   "autoload": {
     "psr-4": {
-      "Neusta\\Pimcore\\EditorConfigBundle\\": ""
+      "Neusta\\Pimcore\\AreabrickConfigBundle\\": ""
```
Comment (jdreesen) [inline composer.json:42]:
```suggestion
      "Neusta\\Pimcore\\AreabrickConfigBundle\\": "src/"
```

File: src/DependencyInjection/Configuration.php [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace Neusta\Pimcore\EditorConfigBundle\DependencyInjection;
+namespace Neusta\Pimcore\AreabrickConfigBundle\src\DependencyInjection;
```
Comment (jdreesen) [inline src/DependencyInjection/Configuration.php:3]: The namespace is wrong everywhere...

File: src/Document/EditableDialogBox/LayoutItem.php [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
 declare(strict_types=1);
 
-namespace Neusta\Pimcore\EditorConfigBundle\Document\EditableDialogBox;
+namespace Neusta\Pimcore\AreabrickConfigBundle\src\Document\EditableDialogBox;
 
 abstract class LayoutItem extends DialogBoxItem
 {
     /**
-     * @param list<DialogBoxItem> $items
+     * @param DialogBoxItem $items
```
Comment (jdreesen) [inline src/Document/EditableDialogBox/LayoutItem.php:9]: That is wrong. It was right before.

File: src/Document/EditableDialogBox/LayoutItem/PanelItem.php [hunk 7]
```diff
@@ -1,17 +1,17 @@
...
+use Neusta\Pimcore\AreabrickConfigBundle\src\Document\EditableDialogBox\DialogBoxItem;
+use Neusta\Pimcore\AreabrickConfigBundle\src\Document\EditableDialogBox\LayoutItem;
 
 class PanelItem extends LayoutItem
 {
     private string $title;
 
     /**
-     * @param list<DialogBoxItem> $items
+     * @param DialogBoxItem $items
```
Comment (jdreesen) [inline src/Document/EditableDialogBox/LayoutItem/PanelItem.php:14]: That is wrong. It was right before.

File: src/Document/EditableDialogBox/LayoutItem/TabPanelItem.php [hunk 8]
```diff
@@ -1,14 +1,14 @@
...
+namespace Neusta\Pimcore\AreabrickConfigBundle\src\Document\EditableDialogBox\LayoutItem;
 
-use Neusta\Pimcore\EditorConfigBundle\Document\EditableDialogBox\LayoutItem;
+use Neusta\Pimcore\AreabrickConfigBundle\src\Document\EditableDialogBox\LayoutItem;
 
 class TabPanelItem extends LayoutItem
 {
     /**
-     * @param list<PanelItem> $items
+     * @param PanelItem $items
```
Comment (jdreesen) [inline src/Document/EditableDialogBox/LayoutItem/TabPanelItem.php:11]: That is wrong. It was right before.

File: src/Installer.php [hunk 9]
```diff
@@ -1,10 +1,8 @@
 <?php declare(strict_types=1);
 
-namespace Neusta\Pimcore\EditorConfigBundle;
+namespace Neusta\Pimcore\AreabrickConfigBundle\src;
 
 use Pimcore\Extension\Bundle\Installer\AbstractInstaller;
-use Pimcore\Model\Asset\Image\Thumbnail;
-use Pimcore\Model\Document\DocType;
 
 class Installer extends AbstractInstaller
```
Comment (jdreesen) [inline src/Installer.php:7]: I don't think we need this. It does nothing.

File: src/DependencyInjection/NeustaPimcoreAreabrickConfigExtension.php [hunk 10]
```diff
@@ -1,22 +1,23 @@
...
         $config = $this->processConfiguration($configuration, $configs);
 
-        $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
+        $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__ . '/../../config'));
         $loader->load('services.yaml');
 
+        /* TODO: ATM bricks.yaml will not be used. May be it could be deleted. */
         if ($config['bricks']) {
             $loader->load('bricks.yaml');
         }
```
Comment (jdreesen) [inline src/DependencyInjection/NeustaPimcoreAreabrickConfigExtension.php:23]: There is no `bricks.yaml`. Delete it.

File: .gitattributes [hunk 11]
```diff
@@ -0,0 +1,10 @@
+/docs/* export-ignore
+/.github export-ignore
+/.gitattributes export-ignore
+/.gitignore export-ignore
+/.php-cs-fixer.php export-ignore
+/phpstan.neon export-ignore
+/phpunit.xml.dist export-ignore
+/run-tests.sh export-ignore
+/tests export-ignore
+/wait-for-it.sh export-ignore
```
Comment (jdreesen) [inline .gitattributes:10]: These files don't exist.

```suggestion
/tests export-ignore
```

File: Document/Base/HasDialogBox.php [hunk 12]
```diff
@@ -1,36 +0,0 @@
...
-
-use Neusta\Pimcore\EditorConfigBundle\Document\EditableDialogBox\DialogBoxBuilder;
-use Pimcore\Extension\Document\Areabrick\EditableDialogBoxConfiguration;
-use Pimcore\Model\Document\Editable;
-use Pimcore\Model\Document\Editable\Area\Info;
-
-/**
- * @template T of DialogBoxBuilder
- */
-trait HasDialogBox
```
Comment (jdreesen) [inline Document/Base/HasDialogBox.php:13]: Why was this removed?

File: Document/EditableDialogBox/EditableItem.php [hunk 13]
```diff
@@ -1,54 +0,0 @@
-<?php
-declare(strict_types=1);
-
-namespace Neusta\Pimcore\EditorConfigBundle\Document\EditableDialogBox;
-
-class EditableItem extends DialogBoxItem
```
Comment (jdreesen) [inline Document/EditableDialogBox/EditableItem.php:6]: Why was this removed?

File: Document/EditableDialogBox/EditableItem/RelationItem.php [hunk 14]
```diff
@@ -1,56 +0,0 @@
-<?php
-declare(strict_types=1);
-
-namespace Neusta\Pimcore\EditorConfigBundle\Document\EditableDialogBox\EditableItem;
-
-use Neusta\Pimcore\EditorConfigBundle\Document\EditableDialogBox\EditableItem;
-
-class RelationItem extends EditableItem
```
Comment (jdreesen) [inline Document/EditableDialogBox/EditableItem/RelationItem.php:8]: Why was this removed?

File: Document/EditableDialogBox/EditableItem/SelectItem.php [hunk 15]
```diff
@@ -1,49 +0,0 @@
-<?php
-declare(strict_types=1);
-
-namespace Neusta\Pimcore\EditorConfigBundle\Document\EditableDialogBox\EditableItem;
-
-use Neusta\Pimcore\EditorConfigBundle\Document\EditableDialogBox\EditableItem;
-
-class SelectItem extends EditableItem
```
Comment (jdreesen) [inline Document/EditableDialogBox/EditableItem/SelectItem.php:8]: Why was this removed?

File: NeustaPimcoreEditorConfigBundle.php [hunk 16]
```diff
@@ -1,38 +0,0 @@
-<?php
-
-namespace Neusta\Pimcore\EditorConfigBundle;
-
-use Pimcore\Extension\Bundle\AbstractPimcoreBundle;
-use Pimcore\Extension\Bundle\Traits\PackageVersionTrait;
-
-class NeustaPimcoreEditorConfigBundle extends AbstractPimcoreBundle
```
Comment (jdreesen) [inline NeustaPimcoreEditorConfigBundle.php:8]: Why was this removed? It's no bundle without it.

**Review by jdreesen — CHANGES_REQUESTED** [review]

File: src/Installer.php (hunk 9 above)
Comment (mike4git) [reply src/Installer.php]: #metoo - removed

File: src/Document/EditableDialogBox/LayoutItem/TabPanelItem.php (hunk 8 above)
Comment (mike4git) [reply src/Document/EditableDialogBox/LayoutItem/TabPanelItem.php]: I don't know what the refactorer did with it but now it's corrected - thx for your strong view.

File: src/Document/EditableDialogBox/LayoutItem/PanelItem.php (hunk 7 above)
Comment (mike4git) [reply src/Document/EditableDialogBox/LayoutItem/PanelItem.php]: Re-done

File: README.md [hunk 17]
```diff
@@ -4,12 +4,17 @@ Adds common techniques to add editor config to areabricks
...
+Require via Composer
+
+```shell
+composer require teamneusta/pimcore-areabrick-config-bundle
+```
+
+As this is a Pimcore bundle, enable and install it
+```shell
+console pimcore:bundle:enable NeustaPimcoreAreabrickConfigBundle
+console pimcore:bundle:install NeustaPimcoreAreabrickConfigBundle
```
Comment (jdreesen) [inline README.md:16]: There is no installer anymore.

**Review by jdreesen — APPROVED** [review]
